src/api/v1/middlewares/validation/userSignup.js

```diff
@@ -1,33 +1,18 @@
 /**
  * Validate the request body for user signup
  */
-
-// api/v1/validators/signupValidator.js
-// You'll need to implement the email uniqueness validation here
-import { pool } from "../../../../config/database.js";
 import Joi from "joi";
-import winstonLogger from "../../../../utils/logger.js";
+import logger from "../../../../utils/logger.js";
+import { getUserbyField } from "../../../../models/UserModel.js";
 export const checkEmailUnique = async (req, res, next) => {
-	const { email } = req.body;
-
-	try {
-		const result = await pool.query("SELECT * FROM users WHERE email = $1", [
-			email,
-		]);
-
-		if (result.rows.length > 0) {
-			return res.status(409).json({ error: "The email is already taken" });
-		}
-
-		next();
-	} catch (error) {
-		winstonLogger.info("Error checking unique email:", error);
-		return res.status(500).json({ error: "Internal server error" });
+	const existingUser = await getUserbyField('email', req.body.email);
```

Comment:
every interaction with the database is done through the Model, in the case, the UserModel. I created a function for querying User by email field. You can use this also for querying userId if you need that in the future. It needs to be modified to accept more than 1 fields, eg. for checking email and password. I will leave that to you to figure out the solution.